hack/monitor_prow.py

```diff
@@ -23,14 +23,15 @@ def main():
     with tempfile.TemporaryDirectory() as log_dir:
         fs = [(display, log_dir), *((highlight, log_dir, x) for x in dcs)]
         with multiprocessing.Pool(len(fs)) as pool:
-            pool.starmap(lambda f, *args: f(*args), fs)
+            for _ in pool.imap_unordered(lambda x: x[0](*x[1:]), fs):
+                pass
```

Comment:
I think the behavior right now (and with this impl?) is to ignore exceptions but not re-execute the worker? This means slowly as they die we get no output from the tool which is kinda bad

Reply:
Iterating over the results has an implicit check for exceptions.  That will
immediately terminate the process and kill all threads (which is fine, in this
case).  E.g. with a synthetic error in one of the `highlight` threads:

```
Traceback (most recent call last):
  File "hack/monitor_prow.py", line 72, in <module>
    main()
  File "hack/monitor_prow.py", line 26, in main
    for _ in pool.imap_unordered(lambda x: x[0](*x[1:]), fs):
  File "/usr/lib64/python3.7/multiprocessing/pool.py", line 748, in next
    raise value
  File "/usr/lib64/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "hack/monitor_prow.py", line 26, in <lambda>
    for _ in pool.imap_unordered(lambda x: x[0](*x[1:]), fs):
  File "hack/monitor_prow.py", line 43, in highlight
    raise Exception()
Exception
```

I've added a comment.

Reply:
As for the change of method, I've tested both version by introducing random
failures and, as far as I can tell, `starmap` first waits for all results, then
iterates over them — which is bad, because our threads never terminate

`imap_unordered` conversely processes each result as it is delivered by the
workers.

Reply:
OK, that sounds perfect